src/Request.ts

```diff
@@ -0,0 +1,350 @@
+	 * the sort information is included in a functional query token to avoid colliding
+	 * with the set of name/value pairs.
+	 */
+	sortParam: string;
+
+	/**
+	 * The target base URL to use for all requests to the server. This string will be
+	 * prepended to the id to generate the URL (relative or absolute) for requests
+	 * sent to the server
+	 */
+	target: string;
+
+	/**
+	 * The indicates if range limits (start and end) should be specified
+	 * a Range header, using items units. If this is set to true, a header
+	 * be included of the form:
+	 * Range: items=start-end
+	 */
+	useRangeHeaders: boolean;
+
+	/**
+	 * This is a basic store for RESTful communicating with a server through JSON
+	 * formatted data. It extends dstore/Store.
+	 * @param options This provides any configuration information that will be mixed into the store
+	 */
+	constructor(options?: RequestStoreArgs) {
+		super(options);
+		/**
+		 * Defaults to JSON, but other formats can be parsed by providing an alternate
+		 * parsing function. If you do want to use an alternate format, you will probably
+		 * want to use an alternate stringify function for the serialization of data as well.
+		 * Also, if you want to support parsing a larger set of JavaScript objects
+		 * outside of strict JSON parsing, you can provide dojo/_base/json.fromJson as the parse function
+		 */
+		this.parse = this.parse || function (data: string) {
+				return JSON.parse(data);
+			};
+		this.headers = this.headers || {};
+		this.target = this.target || '';
+		this._targetContainsQueryString = this.target.lastIndexOf('?') >= 0;
+		this.ascendingPrefix = this.ascendingPrefix || '+';
+		this.descendingPrefix = this.descendingPrefix || '-';
+		this.accepts = this.accepts || 'application/json';
+		this.fetch = this.fetch || function (kwArgs?: dstore.FetchArgs) {
+			const results = this._request(kwArgs);
+			return <dstore.FetchPromise<any>> QueryResults(results.data, {
+				response: results.response
+			});
+		};
+	}
+
+	protected _request(kwArgs: dstore.FetchArgs = {}): {
+		data: dstore.FetchPromise<T>;
+		total: Promise<number>;
+		response: ResponsePromise<T>
+	} {
+		// perform the actual query
+		const RequestHeader: { [ name: string ]: string } = null;
+		const headers = <typeof RequestHeader> lang.mixin(Object.create(this.headers), { Accept: this.accepts });
+
+		if ('headers' in kwArgs) {
+			lang.mixin(headers, kwArgs.headers);
+		}
+
+		const requestUrl = this._renderUrl(kwArgs.queryParams);
+
+		const response = request(requestUrl, {
+			method: 'GET',
+			headers: headers
+		});
+		const collection = this;
+		const parsedResponse = response.then(function (response: Response<string>) {
+			return collection.parse(response.data);
+		});
+		return {
+			data: parsedResponse.then(function (data) {
+				// support items in the results
+				const results = data.items || data;
+				for (let i = 0, l = results.length; i < l; i++) {
+					results[ i ] = collection._restore(results[ i ], true);
```

Comment:
No spaces around indices.  (I am really starting to wonder if the spaces around array literals was really a terrible idea back whenever we came up with it...)